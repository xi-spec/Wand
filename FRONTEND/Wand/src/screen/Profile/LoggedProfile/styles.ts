import { StyleSheet } from 'react-native';
import { COLORS, FONTS, SIZE } from '../../../constants/theme';

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  centerContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: 60
  },

  imageName: {
    color: 'white',
    fontFamily: FONTS.primaryMedium,
    fontSize: SIZE.big
  },

  imageContainer: {
    alignSelf: 'center',
    justifyContent: 'center',
    alignItems: 'center',
    width: 100,
    height: 100,
    borderRadius: 50,
    backgroundColor: COLORS.primary
  },
  locate: {
    paddingVertical: 15,
    flexDirection: 'row'
  },
  welcomeText: {
    paddingTop: 30,
    fontFamily: FONTS.logoRegular,
    fontSize: SIZE.primary
  },
  centerEdit: {
    justifyContent: 'center',
    alignItems: 'center'
  },
  edit: {
    width: '80%',
    height: 60,
    borderRadius: 10,
    flexDirection: 'row',
    borderWidth: 0.0,
    justifyContent: 'space-around',
    alignItems: 'center',
    marginTop: 10,
    shadowColor: 'grey',
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.15,
    shadowRadius: 3.84,

    elevation: 5,
    backgroundColor: 'white',
    marginVertical: 30
  },
  editText: {
    paddingTop: 3,
    fontSize: SIZE.second,
    fontFamily: FONTS.primaryMedium
  },
  icon: {
    marginLeft: 50
  }
});

export default styles;
