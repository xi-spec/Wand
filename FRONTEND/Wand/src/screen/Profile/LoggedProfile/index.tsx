import React from 'react';
import { Text, View, TouchableOpacity } from 'react-native';
import { User } from '../../../models/initiaState';
import { Icon } from 'react-native-elements';
import styles from './styles';

export default function LoggedProfile ({ user, navigation }:
    {user:User,
    navigation:{navigate:Function}}) {
  return (
      <View style={styles.container}>
          <View style={styles.centerContainer}>

          <View style={styles.imageContainer}>
               <Text style={styles.imageName}>{user.name}</Text>
               </View>
          <Text style={styles.welcomeText}> {`Welcome ${user.name}`}</Text>
          {user.address && <View style={styles.locate}>
          <Icon name="locate-outline" type='ionicon' size={20}/>
          <Text > {`${user.address.street} ${user.address.city} ${user.address.postcode} ${user.address.country}`}</Text>
          </View>
          }
          </View>

          <View style={styles.centerEdit}>
              <TouchableOpacity
              style={styles.edit}>
              <Text style={styles.editText}>Edit Profile</Text>
               <Icon style={styles.icon}name="chevron-right" type='evilicon' size={40}/>
              </TouchableOpacity>

              <TouchableOpacity
               testID='navigation'
              style={styles.edit}
              onPress={() => navigation.navigate('Shipping address')}>
              <Text style={styles.editText}>Shopping address</Text>
               <Icon name="chevron-right" type='evilicon' size={40}/>
              </TouchableOpacity>

              <TouchableOpacity
               testID='navigation'
              style={styles.edit}
              onPress={() => navigation.navigate('Payment method')}>
              <Text style={styles.editText}>Payment method</Text>
               <Icon name="chevron-right" type='evilicon' size={40}/>
              </TouchableOpacity>

              <TouchableOpacity
                 testID='navigation'
              style={styles.edit}
                 onPress={() => navigation.navigate('Order history')}>
              <Text style={styles.editText}>Order history</Text>
               <Icon name="chevron-right" type='evilicon' size={40} />
              </TouchableOpacity>
          </View>

      </View>
  );
}
