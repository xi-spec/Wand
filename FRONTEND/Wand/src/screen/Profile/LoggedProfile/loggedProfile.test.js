import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import LoggedProfile from './index';

describe('Given a LoggedProfile component', () => {
  let user;
  let navigation;

  beforeEach(() => {
    user = {
      address: {
        street: '',
        country: '',
        city: '',
        postcode: ''
      },
      name: 'Xi'
    };

    navigation = {
      goBack: jest.fn(),
      navigate: jest.fn()
    };
  });

  describe('When is render', () => {
    test('Then is rendered correctly', () => {
      const tree = render(<LoggedProfile navigation={navigation} user={user}/>);

      expect(tree).toMatchSnapshot();
    });
  });

  describe('When pressed a navigation button', () => {
    test('Then navigation.navigate is invoked', () => {
      const { getAllByTestId } = render(<LoggedProfile navigation={navigation} user={user}/>);

      const navigationArray = getAllByTestId('navigation');

      navigationArray.forEach(navigation => fireEvent.press(navigation));

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });
});
