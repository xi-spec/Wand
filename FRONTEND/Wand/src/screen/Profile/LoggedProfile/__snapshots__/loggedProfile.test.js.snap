// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a LoggedProfile component When is render Then is rendered correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingTop": 60,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#2E3192",
          "borderRadius": 50,
          "height": 100,
          "justifyContent": "center",
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "MontserratMedium",
            "fontSize": 22,
          }
        }
      >
        Xi
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontFamily": "RalewayRegular",
          "fontSize": 18,
          "paddingTop": 30,
        }
      }
    >
       
      Welcome Xi
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "paddingVertical": 15,
        }
      }
    >
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
      <Text>
         
           
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 0,
          "elevation": 5,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-around",
          "marginTop": 10,
          "marginVertical": 30,
          "opacity": 1,
          "shadowColor": "grey",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 3.84,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "MontserratMedium",
            "fontSize": 14,
            "paddingTop": 3,
          }
        }
      >
        Edit Profile
      </Text>
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Object {
              "marginLeft": 50,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 0,
          "elevation": 5,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-around",
          "marginTop": 10,
          "marginVertical": 30,
          "opacity": 1,
          "shadowColor": "grey",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 3.84,
          "width": "80%",
        }
      }
      testID="navigation"
    >
      <Text
        style={
          Object {
            "fontFamily": "MontserratMedium",
            "fontSize": 14,
            "paddingTop": 3,
          }
        }
      >
        Shopping address
      </Text>
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 0,
          "elevation": 5,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-around",
          "marginTop": 10,
          "marginVertical": 30,
          "opacity": 1,
          "shadowColor": "grey",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 3.84,
          "width": "80%",
        }
      }
      testID="navigation"
    >
      <Text
        style={
          Object {
            "fontFamily": "MontserratMedium",
            "fontSize": 14,
            "paddingTop": 3,
          }
        }
      >
        Payment method
      </Text>
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 0,
          "elevation": 5,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-around",
          "marginTop": 10,
          "marginVertical": 30,
          "opacity": 1,
          "shadowColor": "grey",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 3.84,
          "width": "80%",
        }
      }
      testID="navigation"
    >
      <Text
        style={
          Object {
            "fontFamily": "MontserratMedium",
            "fontSize": 14,
            "paddingTop": 3,
          }
        }
      >
        Order history
      </Text>
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
